### PR TITLE
Waiv 2672

### PR DESCRIPTION
**Pull request by oksanana**
Fixes #

### Changes

Summary of changes you made.

* Change 1
* Change 2

### Test plan

Explain how to test changes you made.

* [ ] Step 1
* [ ] Step 2

### Demo (optional)

Sreenshots (before & after) or video that show result of your work.

### CODE REVIEW

**Comment by milklim** [comment]
resolve conflict

File: src/common/routes.js [hunk 1]
```diff
@@ -286,6 +286,11 @@ const routes = [
             exact: true,
             component: HistoryCampaign,
           },
+          {
+            path: '/rewards/guideHistory',
+            exact: true,
+            component: HistoryCampaign,
+          },
```
Comment (milklim) [inline src/common/routes.js:293]: add `'/rewards/(history|guideHistory'|messages)` to parent's path

File: src/client/rewards/CampaignCardHeader/CampaignCardHeader.js [hunk 2]
```diff
@@ -11,7 +11,11 @@ const CampaignCardHeader = ({ intl, campaignData, match }) => {
   const currentUSDPrice = getCurrentUSDPrice();
   const price = get(campaignData, ['objects', '0', 'reward']);
   const isAssigned = get(campaignData, ['objects', '0', 'assigned']);
-  const isMessages = match && match.params.filterKey === 'messages';
+  const isMessages =
+    match &&
+    (match.params.filterKey === 'messages' ||
+      match.params.filterKey === 'history' ||
+      match.params.filterKey === 'guideHistory');
```
Comment (milklim) [inline src/client/rewards/CampaignCardHeader/CampaignCardHeader.js:18]: `messages, history, guideHistory` are not filters

File: src/client/rewards/CampaignFooter/CampaignButtons.js [hunk 3]
```diff
@@ -362,7 +362,10 @@ export default class CampaignButtons extends React.Component {
     const reservationPermlink = get(proposition, ['users', '0', 'permlink']);
     const propositionUserName = get(proposition, ['users', '0', 'name']);
     const reviewPermlink = get(proposition, ['users', '0', 'review_permlink']);
-    const userName = match.params.filterKey === 'messages' ? propositionUserName : user.name;
+    const userName =
+      match.params.filterKey === 'messages' || match.params.filterKey === 'guideHistory'
```
Comment (milklim) [inline src/client/rewards/CampaignFooter/CampaignButtons.js:366]: `messages, history, guideHistory` are not filters

File: src/client/rewards/FilteredRewardsList.js [hunk 4]
```diff
@@ -44,24 +45,36 @@ const FilteredRewardsList = props => {
...
+    sortMessages,
+  );
 
   const showMap = () => dispatch(setMapFullscreenMode(true));
   const IsRequiredObjectWrap =
     !match.params.campaignParent &&
     match.params.filterKey !== 'history' &&
-    match.params.filterKey !== 'messages';
+    match.params.filterKey !== 'messages' &&
+    match.params.filterKey !== 'guideHistory';
```
Comment (milklim) [inline src/client/rewards/FilteredRewardsList.js:69]: `messages, history, guideHistory` are not filters

File: src/client/rewards/Proposition/Proposition.js [hunk 5]
```diff
@@ -48,7 +48,8 @@ const Proposition = ({
   const [isReviewDetails, setReviewDetails] = useState(false);
   const parentObject = getClientWObj(proposition.required_object, usedLocale);
   const requiredObjectName = getFieldWithMaxWeight(proposition.required_object, 'name');
-  const isMessages = match.params.filterKey === 'messages';
+  const isMessages =
+    match.params.filterKey === 'messages' || match.params.filterKey === 'guideHistory';
```
Comment (milklim) [inline src/client/rewards/Proposition/Proposition.js:52]: `messages, history, guideHistory` are not filters

File: src/client/rewards/Proposition/Proposition.js [hunk 6]
```diff
@@ -164,7 +165,8 @@ const Proposition = ({
         {/* changes braked reservation process, changes reverted */}
         {assigned ||
         get(match, ['params', 'filterKey']) === 'history' ||
-        get(match, ['params', 'filterKey']) === 'messages' ? (
+        get(match, ['params', 'filterKey']) === 'messages' ||
+        get(match, ['params', 'filterKey']) === 'guideHistory' ? (
```
Comment (milklim) [inline src/client/rewards/Proposition/Proposition.js:169]: messages, history, guideHistory are not filters

File: src/client/rewards/Rewards.js [hunk 7]
```diff
@@ -197,6 +202,8 @@ class Rewards extends React.Component {
         return this.setState({ sortHistory: sort });
       case 'messages':
         return this.setState({ sortMessages: sort });
+      case 'guideHistory':
+        return this.setState({ sortGuideHistory: sort });
```
Comment (milklim) [inline src/client/rewards/Rewards.js:206]: `messages, history, guideHistory` are not filters

File: src/client/rewards/FilteredRewardsList.js [hunk 8]
```diff
@@ -44,24 +46,36 @@ const FilteredRewardsList = props => {
...
+    sortGuideHistory,
+    sortMessages,
+  );
 
   const showMap = () => dispatch(setMapFullscreenMode(true));
   const IsRequiredObjectWrap =
     !match.params.campaignParent &&
-    match.params.filterKey !== 'history' &&
-    match.params.filterKey !== 'messages';
+    location !== '/rewards/history' &&
```
Comment (LucyHrinishyna) [inline src/client/rewards/FilteredRewardsList.js:68]: '/rewards/history' => ты используешь эти url по несколько раз, вынеси их в переменную

**Pull request merged**